workers/src/main/java/edu/cornell/repository/ConfigParser.java

```diff
@@ -0,0 +1,90 @@
+package edu.cornell.repository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+
+/**
+ * Parses the config file of the repository.
+ */
+public class ConfigParser {
+    private EnumMap<Category, List<String>> configMap = new EnumMap<>(Category.class);
+    private Path configFilePath;
+    public static class ConfigSyntaxException extends Exception {
```

Comment:
I've been putting exceptions in their own files. Should we instead move them as static classes? Doesn't make a difference except in style